docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,552 @@
+-   **Preconditions**: Subnets added to the default or authorized lists cannot already be assigned to a named subnet type (see `update_subnet_type`).
+
+*(Args: `record { who: opt principal; subnets: vec principal; }`, Returns: `()`) *
+
+### `update_subnet_type` (Governance only)
+
+Adds or removes named subnet types (e.g., "Fiduciary", "Storage"). These types provide logical groupings for subnets with specific characteristics.
+
+-   **Caller**: Must be the NNS Governance canister.
+-   **Purpose**:
+    -   **Add**: Creates a new, empty named type. Fails if the type already exists.
+    -   **Remove**: Deletes an existing named type. Fails if the type does not exist or if any subnets are currently assigned to that type.
+
+*(Args: `variant { Add: text; Remove: text; }`, Returns: `variant { Ok: (); Err: text }`)*
+
+### `change_subnet_type_assignment` (Governance only)
+
+Assigns or unassigns specific subnets to a named subnet type.
+
+-   **Caller**: Must be the NNS Governance canister.
+-   **Purpose**:
+    -   **Add**: Assigns a list of subnets to an existing named type. Fails if the type doesn't exist, if any of the subnets are already assigned to *any* type, or if any of the subnets are part of the *default* or per-principal *authorized* lists.
+    -   **Remove**: Unassigns a list of subnets from a specific named type. Fails if the type doesn't exist or if any of the specified subnets are not currently assigned to *that specific* type.
+
+*(Args: `variant { Add: SubnetListWithType; Remove: SubnetListWithType; }` where `SubnetListWithType = record { subnets: vec principal; subnet_type: text; }`, Returns: `variant { Ok: (); Err: text }`)*
+
+
+
+
+## `get_subnet_types_to_subnets`
+
+Returns a mapping from named subnet types (e.g., `"Fiduciary"`) assigned by NNS Governance to the list of subnet principals belonging to that type. Subnets listed here are typically reserved for specific purposes and are separate from the default and authorized lists.
+
+#### Parameters
+
+```candid
+record {}
+```
+
+#### Returns
+
+```candid
+// Current type: SubnetTypesToSubnetsResponse = record { data: vec record { text; vec principal } }
+// Draft type shown for simplicity:
+record {
+  subnets: vec record {
+    subnet_type: text;
+    subnet_ids: vec principal;
+  };
+}
+```
+
+-   Each entry in `subnets` represents a mapping from a subnet type to the list of subnets that belong to that type.
+
+#### Notes
+
+This information can be used by dapps to select appropriate subnets for canister creation when a specific type is desired via `SubnetSelection::Filter`.
+
+
+
+## `get_principals_authorized_to_create_canisters_to_subnets`
+
+Lists principals specifically authorized by NNS Governance to create canisters on non-default subnets, and the corresponding list of subnet principals they are permitted to use.
```

Comment:
I wonder if the relationship between default / non-default subnets (with special permissions) and the "named types" (where any principal can deploy AFAIK) is clear. If we want to be extra clear, maybe we could have a small section introducing these different concepts?